File: online-training-platform/leetcode/75. Sort Colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        map<int, int> m;
        for(int i = 0 ; i< nums.size(); i++){
            m[nums[i]]++;
        }
        map<int, int>::iterator it;
        int i = 0, num;
        vector<int> t;
        for( it = m.begin(); it != m.end() ; it++){
            num = it->second;
            while(num--){
                t.push_back(it->first);
            }
        }
        nums = t;
    }
};

File: online-training-platform/leetcode/75. Sort Colors.kt
class Solution {
    fun sortColors(nums: IntArray): Unit {
        var zeros = 0
        var ones = 0
        var twos = 0

        for(idx in 0 until nums.size) {
            if(nums[idx] == 0){
                zeros += 1
            } else if(nums[idx] == 1) {
                ones += 1
            } else {
                twos += 1
            }
        }

        var idx = 0
        while( zeros > 0){
            nums[idx] = 0
            idx += 1
            zeros -= 1
        }

        while( ones > 0){
            nums[idx] = 1
            idx += 1
            ones -= 1
        }

        while( twos > 0){
            nums[idx] = 2
            idx += 1
            twos -= 1
        }
    }
}